### PR TITLE
feat(kafka): RHICOMPL-3024 parametrize kafka auth protocol/mechanism

### PR DESCRIPTION
**Pull request by skateman**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: config/initializers/0_clowder-config.rb [hunk 1]
```diff
@@ -29,12 +29,13 @@
...
   if kafka_security_protocol
     if kafka_security_protocol == 'sasl'
-      kafka_server_config[:security_protocol] = 'sasl_ssl'
       kafkaCaFile = Tempfile.create
       kafkaCaFile.write(first_kafka_server_config&.dig('cacert'))
       kafka_server_config[:ssl_ca_location] = kafkaCaFile.path
       kafka_server_config[:sasl_username] = first_kafka_server_config&.dig('sasl', 'username')
       kafka_server_config[:sasl_password] = first_kafka_server_config&.dig('sasl', 'password')
+      kafka_server_config[:sasl_mechanism] = first_kafka_server_config&.dig('sasl', 'saslMechanism')
+      kafka_server_config[:security_protocol] = first_kafka_server_config&.dig('sasl', 'securityProtocol')
```
Comment (vkrizan) [inline config/initializers/0_clowder-config.rb:38]: Where do we get the `saslMechanism` and the `securityProtocol` from? I'm unable to find it on the [`KafkaSASLConfig` valid keys](https://github.com/RedHatInsights/clowder-common-ruby/blob/fee91ab094f6ccd91de4eca8d63eb903e1fb90ba/lib/clowder-common-ruby/types.rb#L198-L202). They are, however, set by [Clowder Kafka provider](https://github.com/RedHatInsights/clowder/blob/bbbc4f3f6aa691278603f089af08c136b3f23289/controllers/cloud.redhat.com/config/types.go#L558-L571). I guess the ruby lib deserves this new addition.

/cc @psav

Comment (skateman) [reply config/initializers/0_clowder-config.rb:38]: You need to deploy in ephemeral using `--pool managed-kafka` to see these.

File: config/racecar.rb [hunk 2]
```diff
@@ -3,13 +3,13 @@
   config.group_id_prefix = 'compliance'
 
   if Settings.kafka.security_protocol
-    config.security_protocol = Settings.kafka.security_protocol.try(:to_sym)
+    config.security_protocol = Settings.kafka.security_protocol.try(:downcase).try(:to_sym)
 
     if config.security_protocol == :sasl_ssl
       config.ssl_ca_location = Settings.kafka.ssl_ca_location
```
Comment (vkrizan) [inline config/racecar.rb:9]: Can the CA be also set for security protocol `:ssl`?

Comment (skateman) [reply config/racecar.rb:9]: I need a bit more explanation for this :thinking:

**Review by vkrizan — APPROVED** [review]

**Pull request merged**